src/main/kotlin/lotto/domain/LottoStore.kt

```diff
@@ -2,16 +2,20 @@ package lotto.domain
 
 class LottoStore {
 
-    fun purchaseAuto(money: Int): Lottoes {
-        val quantity = money / LOTTO_COST
+    fun purchaseAuto(money: Money): Lottoes {
+        val quantity = money.currentMoney / LOTTO_COST
+        money.spendAllMoney()
         val lottoes = (1..quantity).map { LottoTicket.generateAuto() }
 
         return Lottoes(lottoes)
     }
 
-    fun purchaseManual(money: Int, numbers: List<Int>): Lottoes {
-        val quantity = money / LOTTO_COST
-        val lottoes = (1..quantity).map { LottoTicket.generateManual(numbers) }
+    fun purchaseManual(money: Money, numberOfManuals: Int, numbersOfTickets: List<List<Int>>): Lottoes {
+        money.spendMoney(LOTTO_COST * numberOfManuals)
+        val lottoes = arrayListOf<LottoTicket>()
+        for (numbers in numbersOfTickets) {
+            lottoes.add(LottoTicket.generateManual(numbers))
+        }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        val lottoes = numbersOfTickets.map { LottoTicket.generateManual(it) }
```